pkg/flow/temporary.go

```diff
@@ -474,7 +473,7 @@ type knativeHandle struct {
 }
 
 func (child *knativeHandle) Run(ctx context.Context) {
-	go child.engine.doActionRequest(ctx, child.ar, child.arReq)
+	child.engine.doActionRequest(ctx, child.ar, child.arReq) // using a go routine may be unsafe (caused panics) where but why?
```

Comment:
@alankm I believe this change is safe. Since Knative/side requests are not handled in an asynchronous manner, I think we should have made this change a long time ago.

Without this change, I was able to crash the engine during the API-E2E test (although due to server recovery and retries in the tests, it was difficult to observe).

Reply:
I think this panic will need further investigation. This function was go-routined for a good reason. For example, the `parallel` and `foreach` states call this function repeatedly with the expectation that it is non-blocking.

Reply:
i have reverted the change. I think the issue rooted from storing a pointer to a map in the context. After fixing this I was unable to reproduce the issue.